File: src/main/java/org/pastafari/javalojban/MonsterContext.java
package org.pastafari.javalojban;

import java.math.BigInteger;
import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public final class MonsterContext {
    public final BigInteger calculationDay;
    public final BigInteger targetDay;
    public String phase;
    public String status;
    public final List<String> branchTrace;
    public final Map<String, BigInteger> metrics;
    public final List<String> diagnostics;

    public MonsterContext(BigInteger calculationDay, BigInteger targetDay) {
        this.calculationDay = calculationDay;
        this.targetDay = targetDay;
        this.phase = "BOOT";
        this.status = "NEW";
        this.branchTrace = new ArrayList<>();
        this.metrics = new LinkedHashMap<>();
        this.diagnostics = new ArrayList<>();
    }
}

File: src/PastafarianCalendar/Monster/MonsterContext.cs
namespace PastafarianCalendar.Monster;

public sealed class MonsterContext
{
    public System.Numerics.BigInteger CalculationDay { get; init; }
    public System.Numerics.BigInteger TargetDay { get; init; }
    public string Phase { get; set; } = "BOOT";
    public string Status { get; set; } = "NEW";
    public int RetryBudget { get; set; }
    public int RecoveryDepth { get; set; }
    public List<string> BranchTrace { get; } = new();
    public Dictionary<string, long> Metrics { get; } = new(StringComparer.Ordinal);
    public List<string> Logs { get; } = new();
    public List<string> Diagnostics { get; } = new();
    public List<string> ValidationFailures { get; } = new();

    public MonsterContext(System.Numerics.BigInteger calculationDay, System.Numerics.BigInteger targetDay)
    {
        CalculationDay = calculationDay;
        TargetDay = targetDay;
    }
}
